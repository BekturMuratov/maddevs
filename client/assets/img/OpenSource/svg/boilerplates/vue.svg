<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>