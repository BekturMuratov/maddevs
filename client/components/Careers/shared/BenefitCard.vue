<template>
  <div class="benefit-card">
    <div class="benefit-card__icon-wrapper">
      <img
        :data-src="$getMediaFromS3(`/images/Careers/svg/${icon.name}.svg`)"
        :width="icon.width"
        :height="icon.height"
        class="benefit-card__icon media_lazy"
      >
    </div>
    <p
      class="benefit-card__title"
      v-html="title"
    >
      {{ title }}
    </p>
  </div>
</template>

<script>
export default {
  name: 'BenefitCard',
  props: {
    icon: {
      type: Object,
      default: () => {},
    },

    title: {
      type: String,
      default: '',
    },
  },
}
</script>

<style lang="scss" scroped>
@import '@/assets/styles/_vars.scss';

.benefit-card {
  padding: 16px;
  background-color: $bgcolor--silver;
  border-radius: 4px;
  &__icon {
    &-wrapper {
      margin-bottom: 16px;
    }
    display: block;
  }
  &-title {
    line-height: 22px;
    letter-spacing: -0.013em;
    color: $text-color--black-oil;
  }
}
</style>
