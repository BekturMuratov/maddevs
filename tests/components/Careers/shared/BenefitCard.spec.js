import BenefitCard from '@/components/Careers/shared/BenefitCard'
import { render, screen } from '@testing-library/vue'

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
}

const props = {
  icon: {
    name: 'trips',
    width: 28,
    height: 32,
  },
  title: 'Onsite business trips',
}

describe('BenefitCard', () => {
  it('is a Vue instance', () => {
    const { container } = render(BenefitCard, {
      mocks,
      props,
    })

    expect(screen.getByText(props.title)).not.toBeNull()
    expect(container).toMatchSnapshot()
  })
})
