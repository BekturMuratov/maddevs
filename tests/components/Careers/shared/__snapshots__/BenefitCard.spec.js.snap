// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BenefitCard is a Vue instance 1`] = `
<div>
  <div
    class="benefit-card"
  >
    <div
      class="benefit-card__icon-wrapper"
    >
      <img
        class="benefit-card__icon media_lazy"
        data-src="img.jpg"
        height="32"
        width="28"
      />
    </div>
     
    <p
      class="benefit-card__title"
    >
      Onsite business trips
    </p>
  </div>
</div>
`;
